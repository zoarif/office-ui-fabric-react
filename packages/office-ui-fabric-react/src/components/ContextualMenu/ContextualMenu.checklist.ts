import { ChecklistStatus } from '../../demo/ComponentStatus/ComponentStatus.types';

export const ContextualMenuStatus = {
  keyboardAccessibilitySupport: ChecklistStatus.unknown,
  markupSupport: ChecklistStatus.unknown,
  highContrastSupport: ChecklistStatus.fail,
  rtlSupport: ChecklistStatus.fail,
  testCoverage: ChecklistStatus.good
};