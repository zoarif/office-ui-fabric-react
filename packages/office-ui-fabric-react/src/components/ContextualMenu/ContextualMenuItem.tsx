import * as React from 'react';
import { hasSubmenu, getIsChecked } from '../../utilities/contextualMenu/index';
import { getRTL } from '../../Utilities';
import { Icon } from '../../Icon';
import { IContextualMenuItemProps } from './ContextualMenuItem.types';

const renderItemIcon = (props: IContextualMenuItemProps) => {
  const {
    item,
    hasIcons,
    classNames
  } = props;

  // Only present to allow continued use of item.icon which is deprecated.
  const { iconProps, icon } = item;

  if (!hasIcons) {
    return null;
  }

  if (item.onRenderIcon) {
    return (
      item.onRenderIcon(props)
    );
  }

  if (iconProps) {
    return <Icon { ...iconProps } className={ classNames.icon } />;
  }

  return <Icon iconName={ icon } className={ classNames.icon } />;
};

const renderCheckMarkIcon = ({ onCheckmarkClick, item, classNames }: IContextualMenuItemProps) => {
  const isItemChecked = getIsChecked(item);
  if (onCheckmarkClick) {
    const onClick = (e: React.MouseEvent<HTMLElement>) => onCheckmarkClick(item, e);

    return (
      <Icon
        iconName={ isItemChecked ? 'CheckMark' : '' }
        className={ classNames.checkmarkIcon }
        onClick={ onClick }
      />
    );
  }
  return null;
};

const renderItemName = ({ item, classNames }: IContextualMenuItemProps) => {
  if (item.name) {
    return <span className={ classNames.label }>{ item.name }</span>;
  }
  return null;
};

const renderSubMenuIcon = ({ item, classNames }: IContextualMenuItemProps) => {
  if (hasSubmenu(item)) {
    return (
      <Icon
        iconName={ getRTL() ? 'ChevronLeft' : 'ChevronRight' }
        { ...item.submenuIconProps }
        className={ classNames.subMenuIcon }
      />
    );
  }
  return null;
};

export const ContextualMenuItem: React.StatelessComponent<IContextualMenuItemProps> = (props) => {
  const { item, classNames } = props;

  return (
    <div
      className={
        item.split ? classNames.linkContentMenu : classNames.linkContent
      }
    >
      { renderCheckMarkIcon(props) }
      { renderItemIcon(props) }
      { renderItemName(props) }
      { renderSubMenuIcon(props) }
    </div>
  );
};
