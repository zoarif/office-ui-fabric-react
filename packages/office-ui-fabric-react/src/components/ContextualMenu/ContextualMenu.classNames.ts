import { memoizeFunction } from '../../Utilities';
import { ITheme, mergeStyleSets } from '../../Styling';
import { getStyles as getContextualMenuStyles, getMenuItemStyles } from './ContextualMenu.styles';
import { IVerticalDividerClassNames } from '../Divider/VerticalDivider.types';
import { getDividerClassNames } from '../Divider/VerticalDivider.classNames';

export interface IContextualMenuClassNames {
  container: string;
  root: string;
  list: string;
  header: string;
  title: string;
}

export interface IMenuItemClassNames {
  item: string;
  divider: string;
  root: string;
  linkContent: string;
  icon: string;
  checkmarkIcon: string;
  subMenuIcon: string;
  label: string;
  splitContainer: string;
  splitPrimary: string;
  splitMenu: string;
  linkContentMenu: string;
}

export const getSplitButtonVerticalDividerClassNames = memoizeFunction((theme: ITheme): IVerticalDividerClassNames => {
  return mergeStyleSets(getDividerClassNames(theme), {
    divider: {
      height: 16,
      width: 1,
    }
  });
});

export const getContextualMenuClassNames = memoizeFunction((
  theme: ITheme,
  className?: string
): IContextualMenuClassNames => {

  const styles = getContextualMenuStyles(theme);

  return mergeStyleSets({
    container: [
      'ms-ContextualMenu-container',
      styles.container,
      className,
      [{
        selectors: {
          ':focus': { outline: 0 }
        }
      }]
    ],
    root: [
      'ms-ContextualMenu is-open',
      styles.root
    ],
    list: [
      'ms-ContextualMenu-list is-open',
      styles.list
    ],
    header: [
      'ms-ContextualMenu-header',
      styles.header
    ],
    title: styles.title
  });
});

export const getItemClassNames = memoizeFunction((
  theme: ITheme,
  disabled: boolean,
  expanded: boolean,
  checked: boolean,
  isAnchorLink: boolean,
  knownIcon: boolean,
  itemClassName?: string,
  dividerClassName?: string,
  iconClassName?: string,
  subMenuClassName?: string,
): IMenuItemClassNames => {

  const styles = getMenuItemStyles(theme);

  return mergeStyleSets({
    item: [
      'ms-ContextualMenu-item',
      styles.item,
      itemClassName,
    ],
    divider: [
      'ms-ContextualMenu-divider',
      styles.divider,
      dividerClassName,
    ],
    root: [
      'ms-ContextualMenu-link',
      styles.root,
      checked && [
        'is-checked',
        styles.rootChecked
      ],
      isAnchorLink && styles.anchorLink,
      expanded && [
        'is-expanded',
        styles.rootExpanded
      ],
      disabled && [
        'is-disabled',
        styles.rootDisabled
      ],
      !disabled && !expanded && [{
        selectors: {
          ':hover': styles.rootHovered,
          ':active': styles.rootPressed,
          '.ms-Fabric--isFocusVisible &:focus, .ms-Fabric--isFocusVisible &:focus:hover': styles.rootFocused,
          '.ms-Fabric--isFocusVisible &:hover': { background: 'inherit;' }
        }
      }],
    ],
    splitPrimary: [
      styles.root,
      checked && [
        'is-checked',
        styles.rootChecked
      ],
      disabled && [
        'is-disabled',
        styles.rootDisabled
      ],
      !disabled && !checked && [{
        selectors: {
          ':hover': styles.rootHovered,
          ':active': styles.rootPressed,
          '.ms-Fabric--isFocusVisible &:focus, .ms-Fabric--isFocusVisible &:focus:hover': styles.rootFocused,
          '.ms-Fabric--isFocusVisible &:hover': { background: 'inherit;' }
        }
      }]
    ],
    splitMenu: [
      styles.root,
      {
        width: 32
      },
      expanded && [
        'is-expanded',
        styles.rootExpanded
      ],
      disabled && [
        'is-disabled',
        styles.rootDisabled
      ],
      !disabled && !expanded && [{
        selectors: {
          ':hover': styles.rootHovered,
          ':active': styles.rootPressed,
          '.ms-Fabric--isFocusVisible &:focus, .ms-Fabric--isFocusVisible &:focus:hover': styles.rootFocused,
          '.ms-Fabric--isFocusVisible &:hover': { background: 'inherit;' }
        }
      }]
    ],
    linkContent: [
      'ms-ContextualMenu-linkContent',
      styles.linkContent
    ],
    linkContentMenu: [
      'ms-ContextualMenu-linkContent',
      styles.linkContent,
      {
        justifyContent: 'center',
      }
    ],
    icon: [
      'ms-ContextualMenu-icon',
      knownIcon && 'ms-ContextualMenu-iconColor ' && styles.iconColor,
      styles.icon,
      iconClassName,
      disabled && [
        'is-disabled',
        styles.iconDisabled
      ]
    ],
    checkmarkIcon: [
      'ms-ContextualMenu-checkmarkIcon',
      knownIcon && 'ms-ContextualMenu-checkmarkIcon ' && styles.checkmarkIcon,
      styles.icon,
      iconClassName,
    ],
    subMenuIcon: [
      'ms-ContextualMenu-submenuIcon',
      styles.subMenuIcon,
      subMenuClassName,
    ],
    label: [
      'ms-ContextualMenu-itemText',
      styles.label
    ],
    splitContainer: [
      styles.splitButtonFlexContainer,
      !disabled && !checked && [{
        selectors: {
          '.ms-Fabric--isFocusVisible &:focus, .ms-Fabric--isFocusVisible &:focus:hover': styles.rootFocused,
        }
      }]
    ]
  });
});