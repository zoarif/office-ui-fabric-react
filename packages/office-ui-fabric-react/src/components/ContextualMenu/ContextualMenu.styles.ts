import {
  FontSizes,
  FontWeights,
  IRawStyle,
  ITheme,
  concatStyleSets,
  getFocusStyle,
  HighContrastSelector
} from '../../Styling';
import { IContextualMenuStyles, IMenuItemStyles } from './ContextualMenu.types';

import { memoizeFunction } from '../../Utilities';

const ContextualMenuItemHeight = '32px';

const getItemHighContrastStyles = memoizeFunction((): IRawStyle => {
  return {
    selectors: {
      [HighContrastSelector]: {
        backgroundColor: 'Highlight',
        borderColor: 'Highlight',
        color: 'HighlightText',
        MsHighContrastAdjust: 'none'
      }
    },
  };
});

export const getMenuItemStyles = memoizeFunction((
  theme: ITheme,
): IMenuItemStyles => {
  const { semanticColors, fonts } = theme;
  const ContextualMenuItemBackgroundHoverColor = semanticColors.menuItemBackgroundHovered;
  const ContextualMenuItemBackgroundSelectedColor = semanticColors.menuItemBackgroundChecked;
  const ContextualMenuItemDividerColor = semanticColors.bodyDivider;

  const menuItemStyles: IMenuItemStyles = {
    item: [
      fonts.medium,
      {
        color: semanticColors.bodyText,
        position: 'relative',
        boxSizing: 'border-box',
      }],
    divider: {
      display: 'block',
      height: '1px',
      backgroundColor: ContextualMenuItemDividerColor,
      position: 'relative'
    },
    root: [
      getFocusStyle(theme),
      fonts.medium,
      {
        color: 'inherit',
        backgroundColor: 'transparent',
        border: 'none',
        width: '100%',
        height: ContextualMenuItemHeight,
        lineHeight: ContextualMenuItemHeight,
        display: 'block',
        cursor: 'pointer',
        padding: '0px 6px',
        textAlign: 'left',
      },
    ],
    rootDisabled: {
      color: semanticColors.disabledBodyText,
      cursor: 'default',
      pointerEvents: 'none',
    },
    rootHovered: {
      backgroundColor: ContextualMenuItemBackgroundHoverColor,
      ...getItemHighContrastStyles()
    },
    rootFocused: {
      backgroundColor: ContextualMenuItemBackgroundHoverColor,
      ...getItemHighContrastStyles()
    },
    rootChecked: {
      ...getItemHighContrastStyles()
    },
    rootPressed: {
      backgroundColor: ContextualMenuItemBackgroundSelectedColor,
      ...getItemHighContrastStyles()
    },
    rootExpanded: {
      backgroundColor: ContextualMenuItemBackgroundSelectedColor,
      color: semanticColors.bodyTextChecked,
      ...getItemHighContrastStyles()
    },
    linkContent: {
      whiteSpace: 'nowrap',
      height: 'inherit',
      display: 'flex',
      alignItems: 'center',
      maxWidth: '100%'
    },
    anchorLink: {
      padding: '0px 6px',
      textRendering: 'auto',
      color: 'inherit',
      letterSpacing: 'normal',
      wordSpacing: 'normal',
      textTransform: 'none',
      textIndent: '0px',
      textShadow: 'none',
      textDecoration: 'none',
      boxSizing: 'border-box'
    },
    label: {
      margin: '0 4px',
      verticalAlign: 'middle',
      display: 'inline-block',
      flexGrow: '1',
      textOverflow: 'ellipsis',
      overflow: 'hidden',
      whiteSpace: 'nowrap'
    },
    icon: {
      display: 'inline-block',
      minHeight: '1px',
      maxHeight: ContextualMenuItemHeight,
      width: '14px',
      margin: '0 4px',
      verticalAlign: 'middle',
      flexShrink: '0',
    },
    iconColor: {
      color: semanticColors.menuIcon,
      selectors: {
        [HighContrastSelector]: {
          color: 'inherit',
        },
        ['$root:hover &']: {
          selectors: {
            [HighContrastSelector]: {
              color: 'HighlightText'
            }
          }
        },
        ['$root:focus &']: {
          selectors: {
            [HighContrastSelector]: {
              color: 'HighlightText'
            }
          }
        }
      }
    },
    iconDisabled: {
      color: semanticColors.disabledBodyText,
    },
    checkmarkIcon: {
      color: semanticColors.bodySubtext,
      selectors: {
        [HighContrastSelector]: {
          color: 'HighlightText',
        }
      }
    },
    subMenuIcon: {
      height: ContextualMenuItemHeight,
      lineHeight: ContextualMenuItemHeight,
      textAlign: 'center',
      display: 'inline-block',
      verticalAlign: 'middle',
      flexShrink: '0',
      fontSize: FontSizes.mini
    },
    splitButtonFlexContainer: [
      getFocusStyle(theme),
      {
        display: 'flex',
        height: ContextualMenuItemHeight,
        flexWrap: 'nowrap',
        justifyContent: 'center',
        alignItems: 'center'
      }],
    splitButtonSeparator: {}
  };

  return concatStyleSets(menuItemStyles);
});

export const getStyles = memoizeFunction((
  theme: ITheme,
): IContextualMenuStyles => {
  const { semanticColors, fonts } = theme;

  const ContextualMenuBackground = semanticColors.bodyBackground;
  const ContextualMenuHeaderColor = semanticColors.menuHeader;

  const styles: IContextualMenuStyles = {
    root: {
      backgroundColor: ContextualMenuBackground,
      minWidth: '180px',
    },
    container: {

    },
    list: {
      listStyleType: 'none',
      margin: '0',
      padding: '0',
      lineHeight: '0',
    },
    title: {
      fontSize: '16px',
      paddingRight: '14px',
      paddingLeft: '14px',
      paddingBottom: '5px',
      paddingTop: '5px',
      backgroundColor: theme.palette.neutralLight
    },
    header: [
      fonts.small,
      {
        fontWeight: FontWeights.semibold,
        color: ContextualMenuHeaderColor,
        background: 'none',
        backgroundColor: 'transparent',
        border: 'none',
        height: ContextualMenuItemHeight,
        lineHeight: ContextualMenuItemHeight,
        cursor: 'default',
        padding: '0px 6px',
        userSelect: 'none',
        textAlign: 'left',
      }
    ],
  };
  return concatStyleSets(styles);
});