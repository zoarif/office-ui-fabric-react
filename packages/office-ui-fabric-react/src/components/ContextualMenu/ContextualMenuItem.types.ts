import { IContextualMenuItem } from './ContextualMenu.types';
import { IMenuItemClassNames } from './ContextualMenu.classNames';

export interface IContextualMenuItemProps
  extends React.HTMLAttributes<IContextualMenuItemProps> {

  /**
   * The item to display
   */
  item: IContextualMenuItem;

  /**
   * Classnames for different aspects of a menu item
   */
  classNames: IMenuItemClassNames;

  /**
   * Index of the item
   */
  index: number;

  /**
   * If this item has icons
   */
  hasIcons: boolean | undefined;

  /**
   * Click handler for the checkmark
   */
  onCheckmarkClick?: ((item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>) => void);
}
