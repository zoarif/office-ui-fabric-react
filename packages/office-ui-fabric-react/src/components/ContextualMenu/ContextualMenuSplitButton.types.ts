
import { IContextualMenuItem } from '../../ContextualMenu';
import { IMenuItemClassNames } from './ContextualMenu.classNames';
import { IContextualMenuItemProps } from './ContextualMenuItem.types';
import { ContextualMenuSplitButton } from './ContextualMenuSplitButton';

export interface IContextualMenuSplitButtonProps extends React.Props<IContextualMenuItem> {
  /**
   * Optional callback to access the ContextualmenuSplitButton interface. Use this instead of ref for accessing
   * the public methods and properties of the component.
   */
  componentRef?: (component: ContextualMenuSplitButton | null) => void;

  /**
   * The item that is used to render the split button.
   */
  item: IContextualMenuItem;

  /**
   * CSS class to apply to the context menu.
   */
  classNames: IMenuItemClassNames;

  /**
   * The index number of the split button among all items in the contextual menu including things like dividers and headers.
   */
  index: number;

  /**
   * The index number of the split button among all items in the contextual menu excluding dividers and headers.
   */
  focusableElementIndex: number;

  /**
   * The total number of items in the contextual menu.
   */
  totalItemCount: number;

  /**
   * Whether or not if the item for the split button uses checkmarks.
   */
  hasCheckmarks?: boolean;

  /**
   * Whether or not the item for the split button uses icons.
   */
  hasIcons?: boolean;

  /**
   * Method to override the render of the individual menu items.
   * @default ContextualMenuItem
   */
  contextualMenuItemAs?: React.ComponentClass<IContextualMenuItemProps> | React.StatelessComponent<IContextualMenuItemProps>;

  /**
   * Callback for when the user's mouse enters the split button.
   */
  onItemMouseEnter?: (item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>, target: HTMLElement) => boolean | void;

  /**
   * Callback for when the user's mouse leaves the split button.
   */
  onItemMouseLeave?: (item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>) => void;

  /**
   * Callback for when the user's mouse moves in the split button.
   */
  onItemMouseMove?: (item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>, target: HTMLElement) => void;

  /**
   * Callback for the mousedown event on the icon button in the split menu.
   */
  onItemMouseDown?: (item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>) => void;

  /**
   * Callback for when the click event on the primary button.
   */
  executeItemClick?: (item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) => void;

  /**
   * Callback for when the click event on the icon button from the split button.
   */
  onItemClick?: (item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) => void;

  /**
   * Callback for when the click event on the icon button which also takes in a specific HTMLElement that will be focused.
   */
  onItemClickBase?: (item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>, target: HTMLElement) => void;

  /**
   * Callback for keyboard events on the split button.
   */
  onItemKeyDown?: (item: IContextualMenuItem, ev: React.KeyboardEvent<HTMLElement>) => void;
}