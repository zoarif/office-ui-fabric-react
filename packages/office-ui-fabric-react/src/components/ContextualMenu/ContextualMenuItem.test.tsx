jest.mock('../../utilities/contextualMenu');

import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ContextualMenuItem } from './ContextualMenuItem';
import { IContextualMenuItemProps } from './ContextualMenuItem.types';
import { IContextualMenuItem } from './ContextualMenu.types';
import { IMenuItemClassNames } from './ContextualMenu.classNames';
import { hasSubmenu } from '../../utilities/contextualMenu';

describe('ContextMenuItemChildren', () => {
  describe('when a checkmark icon', () => {
    let onCheckmarkClick: jest.Mock;
    let menuItem: IContextualMenuItem;
    let menuClassNames: IMenuItemClassNames;
    let wrapper: ShallowWrapper<IContextualMenuItemProps, {}>;

    beforeEach(() => {
      menuItem = { key: '123' };
      menuClassNames = getMenuItemClassNames();
      onCheckmarkClick = jest.fn();

      wrapper = shallow(
        <ContextualMenuItem
          item={ menuItem }
          classNames={ menuClassNames }
          index={ 1 }
          hasIcons={ undefined }
          onCheckmarkClick={ onCheckmarkClick }
        />
      );
    });

    it('renders the component with the checkmark', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('when the checkmark is clicked', () => {
      let event: jest.Mock;
      beforeEach(() => {
        event = jest.fn();
        wrapper.find('.checkmarkIcon').simulate('click', event);
      });

      it('invokes the onCheckmarkClick callback', () => {
        expect(onCheckmarkClick).toHaveBeenCalledWith(menuItem, event);
      });
    });
  });

  describe('when it has icons', () => {
    describe('when it has iconProps', () => {
      let menuItem: IContextualMenuItem;
      let menuClassNames: IMenuItemClassNames;
      let wrapper: ShallowWrapper<IContextualMenuItemProps, {}>;

      beforeEach(() => {
        menuItem = { key: '123', icon: 'itemIcon', name: 'menuItem' };
        menuClassNames = getMenuItemClassNames();

        wrapper = shallow(
          <ContextualMenuItem
            item={ menuItem }
            classNames={ menuClassNames }
            index={ 1 }
            hasIcons={ true }
          />
        );
      });

      it('renders the icon', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('when it doesnt have iconProps', () => {
      let menuItem: IContextualMenuItem;
      let menuClassNames: IMenuItemClassNames;
      let wrapper: ShallowWrapper<IContextualMenuItemProps, {}>;

      beforeEach(() => {
        menuItem = { key: '123', iconProps: {} };
        menuClassNames = getMenuItemClassNames();

        wrapper = shallow(
          <ContextualMenuItem
            item={ menuItem }
            classNames={ menuClassNames }
            index={ 1 }
            hasIcons={ true }
          />
        );
      });

      it('renders the icon with iconName', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('when it has a sub menu', () => {
    let menuItem: IContextualMenuItem;
    let menuClassNames: IMenuItemClassNames;
    let wrapper: ShallowWrapper<IContextualMenuItemProps, {}>;

    beforeEach(() => {
      (hasSubmenu as jest.Mock).mockReturnValue(true);
      menuItem = { key: '123', iconProps: {}, submenuIconProps: {} };
      menuClassNames = getMenuItemClassNames();

      wrapper = shallow(
        <ContextualMenuItem
          item={ menuItem }
          classNames={ menuClassNames }
          index={ 1 }
          hasIcons={ true }
        />
      );
    });

    it('renders the menu icon', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

function getMenuItemClassNames(): IMenuItemClassNames {
  return {
    item: 'item',
    divider: '---',
    root: 'root',
    linkContent: 'linkContent',
    icon: 'icon',
    checkmarkIcon: 'checkmarkIcon',
    subMenuIcon: 'subMenuIcon',
    label: 'label',
    splitContainer: 'splitContainer',
    splitPrimary: 'splitPrimary',
    splitMenu: 'splitMenu',
    linkContentMenu: 'linkContentMenu',
  };
}
