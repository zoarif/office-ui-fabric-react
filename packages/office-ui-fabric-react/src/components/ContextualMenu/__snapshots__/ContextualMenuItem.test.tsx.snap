// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenuItemChildren when a checkmark icon renders the component with the checkmark 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classNames={
      Object {
        "checkmarkIcon": "checkmarkIcon",
        "divider": "---",
        "icon": "icon",
        "item": "item",
        "label": "label",
        "linkContent": "linkContent",
        "linkContentMenu": "linkContentMenu",
        "root": "root",
        "splitContainer": "splitContainer",
        "splitMenu": "splitMenu",
        "splitPrimary": "splitPrimary",
        "subMenuIcon": "subMenuIcon",
      }
    }
    hasIcons={undefined}
    index={1}
    item={
      Object {
        "key": "123",
      }
    }
    onCheckmarkClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          className="checkmarkIcon"
          iconName=""
          onClick={[Function]}
        />,
        null,
        null,
        null,
      ],
      "className": "linkContent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "checkmarkIcon",
          "iconName": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            className="checkmarkIcon"
            iconName=""
            onClick={[Function]}
          />,
          null,
          null,
          null,
        ],
        "className": "linkContent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "checkmarkIcon",
            "iconName": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ContextMenuItemChildren when it has a sub menu renders the menu icon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classNames={
      Object {
        "checkmarkIcon": "checkmarkIcon",
        "divider": "---",
        "icon": "icon",
        "item": "item",
        "label": "label",
        "linkContent": "linkContent",
        "linkContentMenu": "linkContentMenu",
        "root": "root",
        "splitContainer": "splitContainer",
        "splitMenu": "splitMenu",
        "splitPrimary": "splitPrimary",
        "subMenuIcon": "subMenuIcon",
      }
    }
    hasIcons={true}
    index={1}
    item={
      Object {
        "iconProps": Object {},
        "key": "123",
        "submenuIconProps": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <Icon
          className="icon"
        />,
        null,
        <Icon
          className="subMenuIcon"
          iconName="ChevronRight"
        />,
      ],
      "className": "linkContent",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "icon",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "subMenuIcon",
          "iconName": "ChevronRight",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Icon
            className="icon"
          />,
          null,
          <Icon
            className="subMenuIcon"
            iconName="ChevronRight"
          />,
        ],
        "className": "linkContent",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "icon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "subMenuIcon",
            "iconName": "ChevronRight",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ContextMenuItemChildren when it has icons when it doesnt have iconProps renders the icon with iconName 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classNames={
      Object {
        "checkmarkIcon": "checkmarkIcon",
        "divider": "---",
        "icon": "icon",
        "item": "item",
        "label": "label",
        "linkContent": "linkContent",
        "linkContentMenu": "linkContentMenu",
        "root": "root",
        "splitContainer": "splitContainer",
        "splitMenu": "splitMenu",
        "splitPrimary": "splitPrimary",
        "subMenuIcon": "subMenuIcon",
      }
    }
    hasIcons={true}
    index={1}
    item={
      Object {
        "iconProps": Object {},
        "key": "123",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <Icon
          className="icon"
        />,
        null,
        null,
      ],
      "className": "linkContent",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "icon",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Icon
            className="icon"
          />,
          null,
          null,
        ],
        "className": "linkContent",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "icon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ContextMenuItemChildren when it has icons when it has iconProps renders the icon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classNames={
      Object {
        "checkmarkIcon": "checkmarkIcon",
        "divider": "---",
        "icon": "icon",
        "item": "item",
        "label": "label",
        "linkContent": "linkContent",
        "linkContentMenu": "linkContentMenu",
        "root": "root",
        "splitContainer": "splitContainer",
        "splitMenu": "splitMenu",
        "splitPrimary": "splitPrimary",
        "subMenuIcon": "subMenuIcon",
      }
    }
    hasIcons={true}
    index={1}
    item={
      Object {
        "icon": "itemIcon",
        "key": "123",
        "name": "menuItem",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <Icon
          className="icon"
          iconName="itemIcon"
        />,
        <span
          className="label"
        >
          menuItem
        </span>,
        null,
      ],
      "className": "linkContent",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "icon",
          "iconName": "itemIcon",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "menuItem",
          "className": "label",
        },
        "ref": null,
        "rendered": "menuItem",
        "type": "span",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Icon
            className="icon"
            iconName="itemIcon"
          />,
          <span
            className="label"
          >
            menuItem
          </span>,
          null,
        ],
        "className": "linkContent",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "icon",
            "iconName": "itemIcon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "menuItem",
            "className": "label",
          },
          "ref": null,
          "rendered": "menuItem",
          "type": "span",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
